src/app/components/toast-container/toast-container.component.ts

```diff
@@ -11,22 +21,24 @@ import { ToastEvent, ToastEventsWithIndex } from '../../models/toast.types';
 export class ToastContainerComponent implements OnInit {
   toastStack: ToastEventsWithIndex[] = [];
 
+  toastEvents$ = this.toastService.toastEvents$;
+
+  @ViewChild('toast')
+  toastComponent!: ElementRef;
+
+  // @ViewChildren('toastsRef')
+  // toastsRef!: QueryList<ElementRef>;
+
+  @ViewChildren('toasts')
+  toastComp!: QueryList<ElementRef>;
+
   constructor(private toastService: ToastService) {}
 
-  ngOnInit(): void {
-    this.subscribeToToasts();
-  }
+  ngOnInit(): void {}
 
-  private subscribeToToasts() {
-    let eventIndex = -1;
-    this.toastService.toastEvents$.subscribe((toast) => {
-      eventIndex++;
-      const currentToast: ToastEvent = toast;
-      this.toastStack.push({ event: currentToast, index: eventIndex });
-    });
-  }
+  disposeToast() {}
 
-  disposeToast(index: number) {
-    this.toastStack = this.toastStack.filter((toast) => toast.index != index);
```

Comment:
Need help understanding what the nice way to dispose toasts would be here?

can we have the index from ngfor, as templateRef for the multiple <app-toast> that will be generated? and use that with viewChildren to remove element that matches the index?